CONTRIBUTING.md

```diff
@@ -1,95 +1,151 @@
 
-Then please do not report your issue here - you should instead report it to [https://support.docker.com](https://support.docker.com)
+Please refer to [Security Guide](SECURITY.md) on how to report security vulnerabilities.
 
-### If you...
+## Reporting an Issue
 
- - need help setting up notary
- - can't figure out something
- - are not sure what's going on or what your problem is
+Reporting an issue is a great way to contribute. Please open an issue on Github with required information.
 
-Then please do not open an issue here yet - you should first try one of the following support forums:
+Prior to opening a new issue, we kindly request that you check the existing issues to prevent duplicate submissions. In the event that you come across an issue that aligns with your problem please do following:
 
- - irc: #docker-trust on freenode
+- Carefully review the issue description and comments to validate that your issue aligns with a similar problem. It's important to note that a specific error could potentially be associated with various issues. Additionally, you might discover a potential solution or workaround in the comments section
+- To receive notifications regarding an issue, just click the 'subscribe' button
+- Please only comment if you have new, technical, and relevant information to add to the case.
 
-## Reporting an issue properly
+Issues will be marked with different labels for different purpose, see [Labels](#labels) for details.
 
-By following these simple rules you will get better and faster feedback on your issue.
+### Issue Lifecycle
 
- - search the bugtracker for an already reported issue
+The issue lifecycle is mainly driven by the maintainers.
 
-### If you found an issue that describes your problem:
+1. Issue creation:
+   - After an issue is created, it will be labelled `bug` or `enhancement`, and `triage`.
+2. Triage: The maintainers will triage the issues on a regular basis. The results of triage will be one of the following:
+   - Remove label `triage`, and add the issues to the correct milestone.
+   - Remove the `triage` label and apply the `question` label to indicate that this issue requires additional information. Once this issue has been clarified with the necessary details, it can then be assigned to the appropriate milestone or labeled as `wont fix`.
+   - Remove the `triage` label and assign the `wont fix` label to signify that this issue falls outside the scope of the Notary Project or for other relevant reasons.
+3. Development:
+   - An issue labelled `enhancement` normally require solution proposal and spec updates if needed before implementation
+   - An issue that is labeled as `enhancement` or `bug` should be connected to the PR that resolves it
+4. Issue closure: Follow the [guidance](https://docs.github.com/en/issues/tracking-your-work-with-issues/closing-an-issue)
+   - Close as completed once PRs are merged.
+   - Close as completed once the author accepts the `wont fix` decision.
```

Comment:
If the author does not reply, the issue will become stale, then it will be closed as not planned eventually. If the author does not accept the decision, then this issue will not be closed, and the discussion continues in the issue. I updated the description according to GitHub guidance for simplicity.